packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -661,3 +661,124 @@ test.describe('actions', () => {
+                props: {
+                    isOpen: true,
+                    isDismissible: true,
+                    isLoading: true,
+                    hasBackButton: true,
+                    aria: {
+                        close: 'Close label info',
+                        back: 'Back label info',
+                        loading: 'Loading label info',
+                    },
+                },
+            });
+
+            // Act
+            // Close button
+            const closeButton = await component.locator(closeButtonSelector);
+            const ariaCloseLabel = await closeButton.getAttribute('aria-label');
+
+            // Back button
+            const backButton = await component.locator(backButtonSelector);
+            const ariaBackLabel = await backButton.getAttribute('aria-label');
+
+            // Assert
+            await expect(ariaCloseLabel).toBe('Close label info');
+            await expect(ariaBackLabel).toBe('Back label info');
+        });
+
+        test.describe('when modal `isloading` is true', () => {
+            test('should render component with the correct aria values: `aria-label` & `aria-busy`', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieModal, {
+                    props: {
+                        isOpen: true,
+                        isLoading: true,
+                        aria: {
+                            loading: 'Loading label info',
+                        },
+                    },
+                });
+
+                // Loading state
+                const pieModalComponent = await component.locator(componentSelector);
+                const ariaLoadingLabel = await pieModalComponent.getAttribute('aria-label');
+                const ariaLoadingBusy = await pieModalComponent.getAttribute('aria-busy');
+
+                // Assert
+                await expect(ariaLoadingLabel).toBe('Loading label info');
+                await expect(ariaLoadingBusy).toBe('true');
+            });
+        });
+    });
+
+    test.describe('when aria does not exist', () => {
+        test('should not render the aria-labels', async ({ mount }) => {
+            // Arrange
+            const component = await mount(PieModal, {
+                props: {
+                    isOpen: true,
+                    isDismissible: true,
+                    hasBackButton: true,
+                },
+            });
+
+            // Act
+            // Close button
+            const closeButton = await component.locator(closeButtonSelector);
+            const ariaCloseLabel = await closeButton.getAttribute('aria-label');
+
+            // Back button
+            const backButton = await component.locator(backButtonSelector);
+            const ariaBackLabel = await backButton.getAttribute('aria-label');
+
+            // Assert
+            await expect(ariaCloseLabel).toBe(null);
+            await expect(ariaBackLabel).toBe(null);
```

Comment:
are you able to test updating the loading prop after mounting and re-checking the attribute?

I.e. start with loading, then update to false, then check attribute

Reply:
Do you mean in the next test as this one isn't about loading? I can add that but unsure of the benefits as there is a test checking the loading state. The state in the next test would fail if the attribute existed so i think mounting it and unmounting it might be a bit of overkill in this situation?

Reply:
Basically are we able to test that the aria-busy attribute dynamically changes when the state changes, rather than just at the time the component is mounted

Reply:
Have added the suggestion in.